### ISSUE TITLE
Microsoft.AspNetCore.OpenApi v9.0.0-preview.6.24328.4 is generating schemas for primitive types

### ISSUE THREAD

**Issue by michael-wolfenden** [dotnet/aspnetcore#56735]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

The following program

```csharp
var builder = WebApplication.CreateBuilder(args);
builder.Services.AddOpenApi();

var app = builder.Build();
app.MapOpenApi();

app.MapPost("/article", (CreatePostRequest request) => request);

app.Run();

record CreatePostRequest(
    string stringProperty,
    string? nullableStringProperty,
    int intPropery,
    int? nullableIntProperty);
```

Is generating the following OpenApi specification with schema references for each of the primitive type, including one with an invalid name - `int?`. 

```json
{
  "openapi": "3.0.1",
  "info": {
    "title": "WebApplication1 | v1",
    "version": "1.0.0"
  },
  "paths": {
    "/article": {
      "post": {
        "tags": [
          "WebApplication1"
        ],
        "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/CreatePostRequest"
              }
            }
          },
          "required": true
        },
        "responses": {
          "200": {
            "description": "OK",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/CreatePostRequest"
                }
              }
            }
          }
        }
      }
    }
  },
  "components": {
    "schemas": {
      "CreatePostRequest": {
        "required": [
          "stringProperty",
          "nullableStringProperty",
          "intPropery",
          "nullableIntProperty"
        ],
        "type": "object",
        "properties": {
          "stringProperty": {
            "$ref": "#/components/schemas/string"
          },
          "nullableStringProperty": {
            "$ref": "#/components/schemas/string2"
          },
          "intPropery": {
            "$ref": "#/components/schemas/int"
          },
          "nullableIntProperty": {
            "$ref": "#/components/schemas/int?"
          }
        }
      },
      "int": {
        "type": "integer",
        "format": "int32"
      },
      "int?": {
        "type": "integer",
        "format": "int32",
        "nullable": true
      },
      "string": {
        "type": "string"
      },
      "string2": {
        "type": "string",
        "nullable": true
      }
    }
  },
  "tags": [
    {
      "name": "WebApplication1"
    }
  ]
}
```

### Expected Behavior

The equivalent using `Swashbuckle.AspNetCore `

```csharp
var builder = WebApplication.CreateBuilder(args);
builder.Services.AddEndpointsApiExplorer();
builder.Services.AddSwaggerGen(x => x.SupportNonNullableReferenceTypes());

var app = builder.Build();
app.UseSwagger();

app.MapPost("/article", (CreatePostRequest request) => request);

app.Run();

record CreatePostRequest(
    string stringProperty,
    string? nullableStringProperty,
    int intPropery,
    int? nullableIntProperty);
```

Generates the following OpenApi specification with the primitive types inlined

```json
{
  "openapi": "3.0.1",
  "info": {
    "title": "WebApplication1",
    "version": "1.0"
  },
  "paths": {
    "/article": {
      "post": {
        "tags": [
          "WebApplication1"
        ],
        "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/CreatePostRequest"
              }
            }
          },
          "required": true
        },
        "responses": {
          "200": {
            "description": "OK",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/CreatePostRequest"
                }
              }
            }
          }
        }
      }
    }
  },
  "components": {
    "schemas": {
      "CreatePostRequest": {
        "type": "object",
        "properties": {
          "stringProperty": {
            "type": "string"
          },
          "nullableStringProperty": {
            "type": "string",
            "nullable": true
          },
          "intPropery": {
            "type": "integer",
            "format": "int32"
          },
          "nullableIntProperty": {
            "type": "integer",
            "format": "int32",
            "nullable": true
          }
        },
        "additionalProperties": false
      }
    }
  }
}
```

### Steps To Reproduce

See above

### .NET Version

9.0.100-preview.6.24328.19

**Comment by captainsafia** [comment]
@michael-wolfenden Thanks for filing this issue!

This is resolved in preview.7. Would you mind trying those bits? Let me know if you need help using the preview.7 nightlies.

**Comment by michael-wolfenden** [comment]
@captainsafia Confirmed that this is resolved in `v9.0.0-preview.7.24360.4`
